integration/spark/src/main/common/java/io/openlineage/spark/agent/OpenLineageSparkListener.java

```diff
@@ -140,8 +141,13 @@ public void onOtherEvent(SparkListenerEvent event) {
 
   /** called by the SparkListener when a spark-sql (Dataset api) execution starts */
   private static void sparkSQLExecStart(SparkListenerSQLExecutionStart startEvent) {
+    if (SQLExecution.getQueryExecution(startEvent.executionId()) == null) {
+      return;
+    }
```

Comment:
No point in adding this here - the manipulation of the underlying map happens in a different thread. It may well be visible here, but gone by the time we try to get it in the `ContextFactory`. Remove this and just rely on the `if (context != null)` check below.

Reply:
When I remove this, the integration test `spark_write_delta_table_version.py` fails due to the exception:
```
java.lang.NullPointerException
        at io.openlineage.spark.agent.lifecycle.ContextFactory.createSparkSQLExecutionContext(ContextFactory.java:43)
```
This looks like delta triggers Spark listener event with `executionId` that won't work within `SQLExecution.getQueryExecution(executionId)`. That's why the check is added here.